crypto/asn1/a_verify.c

```diff
@@ -180,8 +180,9 @@ int ASN1_item_verify_ctx(const ASN1_ITEM *it, const X509_ALGOR *alg,
             if (mdnid != NID_undef) {
                 type = EVP_get_digestbynid(mdnid);
                 if (type == NULL) {
-                    ERR_raise(ERR_LIB_ASN1,
-                              ASN1_R_UNKNOWN_MESSAGE_DIGEST_ALGORITHM);
+                    ERR_raise_data(ERR_LIB_ASN1,
+                                   ASN1_R_UNKNOWN_MESSAGE_DIGEST_ALGORITHM,
+                                   "nid=0x%x", mdnid);
```

Comment:
Using the algorithm's name would be better IMO.
The project is moving away from using NIDs and it's better to have the _proper_ fix in place than update again later.

Reply:
In my case it would not have helped as there was no name. The ASN1 code grabbed the NID from the certificate and tried to find EVP_MD for it which failed (`EVP_get_digestbyname()`, which the `EVP_get_digestbynid()` expands to, for the object returned NULL), because my OpenSSL build did not call `EVP_add_digest()`.